Google authentication, session management, initial routing - AB#42

Kesuera opened the pull request:
I have added google authentication and session management, link to library: https://github.com/sidebase/nuxt-auth. At the same time, some primitive routing was added. Part about our team - non-authorized, part for the project - authorized.

Kesuera committed:
Google authentication, session management, initial routing - AB#42

bran0h commented on pages/index.vue:
```diff
@@ -0,0 +1,19 @@
+<template>
+  <div>
+    <h1>Home page</h1>
+    <button @click="goToPage('/team/about')">Team section</button>
+    <button @click="goToPage('/project/login')">Project section</button>
```
Router instance is auto-injected in nuxt, you can use `$router.push(...)` instead of custom function

bran0h commented on layouts/team.vue:
I don't think this layout is necessary, we can use default layout for team instead

bran0h commented on pages/index.vue:
```diff
@@ -0,0 +1,19 @@
+
+<script lang="ts" setup>
+definePageMeta({
+  auth: false,
```
I would move this meta to default layout

bran0h commented on pages/team/about.vue:
```diff
@@ -0,0 +1,12 @@
+
+<script lang="ts" setup>
+definePageMeta({
+  auth: false,
+  layout: "team",
```
As mentioned earlier

bran0h reviewed (changes requested):
Please add `.env.example` file, so we know which variables should be provided

Kesuera committed:
Changed layout and added .env.example

Kesuera commented:
I added a .env.example file, changed routing to use $router.push(...) and made some layout changes => default layout is for project, team layout is preserved and default index page is the team dashboard page from which user can move to project section.
